virtual_rainforest/models/soil/soil_model.py

```diff
@@ -132,29 +147,122 @@ def spinup(self) -> None:
+        """Replace soil pools with previously calculated new pool values.
+
+        The state of particular soil pools will effect the rate of other processes in
+        the soil module. These processes in turn can effect the exchange rates between
+        the original soil pools. Thus, a separate update function is necessary so that
+        the new values for all soil module components can be calculated on a single
+        state, which is then updated (using this function) when all values have been
+        calculated.
+
+        Args:
+            new_pools: Array of new pool values to insert into the data object
+        """
+
+        self.data["low_molecular_weight_c"] = new_pools["new_lmwc"]
+        self.data["mineral_associated_om"] = new_pools["new_maom"]
+
+    def integrate_soil_model(self) -> Dataset:
+        """Function to integrate the soil model.
+
+        For now a single integration will be used to advance the entire soil module.
+        However, this might get split into several separate integrations in future (if
+        that is feasible).
+
+        This function unpacks the variables that are to be integrated into a single
+        numpy array suitable for integration. This order of this numpy array is as
+        follows: [low_molecular_weight_c, mineral_associated_om].
+
+        Returns:
+            A data array containing the new pool values (i.e. the values at the final
+            time point)
+
+        Raises:
+            IntegrationError: When the integration cannot be successfully completed.
+        """
+
+        # Find number of grid cells integration is being performed over
+        no_cells = self.data.grid.n_cells
+
+        # Extract update interval
+        update_time = self.update_interval.astype("timedelta64[s]").astype(float)
+        # Convert from second to day units
+        t_span = (0.0, update_time / (60.0 * 60.0 * 24.0))
+
+        # Construct vector of initial values y0
+        y0 = np.concatenate(
+            [
+                self.data["low_molecular_weight_c"].to_numpy(),
+                self.data["mineral_associated_om"].to_numpy(),
+            ]
+        )
+
+        # Carry out simulation
+        output = solve_ivp(
+            construct_full_soil_model,
+            t_span,
+            y0,
+            args=(
+                self.data,
+                no_cells,
+            ),
+        )
+
+        # Check whether or not integration succeeded
+        if output.success:
+            return Dataset(
+                data_vars=dict(
+                    new_lmwc=DataArray(output.y[:no_cells, -1], dims="cell_id"),
+                    new_maom=DataArray(output.y[no_cells:, -1], dims="cell_id"),
+                )
+            )
+        else:
+            LOGGER.error(
+                "Integration of soil module failed with following message: %s"
+                % str(output.message)
+            )
+            raise IntegrationError()
```

Comment:
Seeing as both of these blocks exit the function scope, you don't need the else part. I'd also write the error condition first for clarity:

```suggestion
        # Check if integration failed
        if not output.success:
            LOGGER.error(
                "Integration of soil module failed with following message: %s"
                % str(output.message)
            )
            raise IntegrationError()

        return Dataset(
            data_vars=dict(
                new_lmwc=DataArray(output.y[:no_cells, -1], dims="cell_id"),
                new_maom=DataArray(output.y[no_cells:, -1], dims="cell_id"),
            )
        )
```